stdlib/3/tkinter/__init__.pyi

```diff
@@ -11,7 +12,69 @@ READABLE: Any
 WRITABLE: Any
 EXCEPTION: Any
 
-class Event: ...
+if sys.version_info >= (3, 6):
+    class EventType(str, Enum):
+        Activate: str = ...
+        ButtonPress: str = ...
+        ButtonRelease: str = ...
+        Circulate: str = ...
+        CirculateRequest: str = ...
+        ClientMessage: str = ...
+        Colormap: str = ...
+        Configure: str = ...
+        ConfigureRequest: str = ...
+        Create: str = ...
+        Deactivate: str = ...
+        Destroy: str = ...
+        Enter: str = ...
+        Expose: str = ...
+        FocusIn: str = ...
+        FocusOut: str = ...
+        GraphicsExpose: str = ...
+        Gravity: str = ...
+        KeyPress: str = ...
+        KeyRelease: str = ...
+        Keymap: str = ...
+        Leave: str = ...
+        Map: str = ...
+        MapRequest: str = ...
+        Mapping: str = ...
+        Motion: str = ...
+        MouseWheel: str = ...
+        NoExpose: str = ...
+        Property: str = ...
+        Reparent: str = ...
+        ResizeRequest: str = ...
+        Selection: str = ...
+        SelectionClear: str = ...
+        SelectionRequest: str = ...
+        Unmap: str = ...
+        VirtualEvent: str = ...
+        Visibility: str = ...
+
+class Event:
+    serial: int
+    num: int
+    focus: bool
+    height: int
+    width: int
+    keycode: int
+    state: Union[int, str]
```

Comment:
How did you get the type for `state`? I'm not familiar with this module, but it might be just int: https://github.com/python/cpython/blob/d31b376bc85dded9d059259737b204e544707d07/Lib/tkinter/__init__.py#L1575
(note that calling `repr` on an Event will display the int by an or of bitmasks)

Reply:
For events of type `<Visibility>`, `state` is a `str` (examples are `VisibilityFullyObscured` and `VisibilityUnobscured`). However for events of type `<ButtonPress>`, it is an `int`.